src/VirtualClient/VirtualClient.Actions/BlenderBenchmark/BlenderBenchmarkExecutor.cs

```diff
@@ -0,0 +1,269 @@
+        private IFileSystem fileSystem;
+        private ISystemManagement systemManagement;
+
+        /// <summary>
+        /// Constructor for <see cref="BlenderBenchmarkExecutor"/>
+        /// </summary>
+        /// <param name="dependencies">Provides required dependencies to the component.</param>
+        /// <param name="parameters">Parameters defined in the profile or supplied on the command line.</param>
+        public BlenderBenchmarkExecutor(IServiceCollection dependencies, IDictionary<string, IConvertible> parameters)
+            : base(dependencies, parameters)
+        {
+            this.fileSystem = dependencies.GetService<IFileSystem>();
+            this.systemManagement = dependencies.GetService<ISystemManagement>();
+        }
+
+        /// <summary>
+        /// The Scenes that will be run by Blender Benchmark.
+        /// </summary>
+        public string[] Scenes
+        {
+            get
+            {
+                // Trim and remove any whitespaces
+                return this.Parameters.GetValue<string>(nameof(BlenderBenchmarkExecutor.Scenes)).Split(",", StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+            }
+        }
+
+        /// <summary>
+        /// The Blender Version being used by the blender benchmark.
+        /// </summary>
+        public string BlenderVersion
+        {
+            get
+            {
+                return this.Parameters.GetValue<string>(nameof(BlenderBenchmarkExecutor.BlenderVersion));
+            }
+        }
+
+        /// <summary>
+        /// The device types to be tested on.
+        /// </summary>
+        public string[] DeviceTypes
+        {
+            get
+            {
+                // Trim and remove any whitespaces
+                return this.Parameters.GetValue<string>(nameof(BlenderBenchmarkExecutor.DeviceTypes)).Split(",", StringSplitOptions.RemoveEmptyEntries | StringSplitOptions.TrimEntries);
+            }
+        }
+
+        /// <summary>
+        /// The path to the blender-benchmark-cli.exe.
+        /// </summary>
+        public string ExecutablePath { get; set; }
+
+        /// <summary>
+        /// Defines the path to the Blender Benchmark package that contains the workload
+        /// executable.
+        /// </summary>
+        protected DependencyPath Package { get; set; }
+
+        /// <summary>
+        /// Initializes the environment
+        /// </summary>
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            PlatformSpecifics.ThrowIfNotSupported(this.Platform);
+
+            await this.InitializePackageLocationAsync(cancellationToken)
+                .ConfigureAwait(false);
+
+            this.ExecutablePath = this.PlatformSpecifics.Combine(this.Package.Path, BlenderBenchmarkExecutableName);
+
+            await this.InitializeBlenderBenchmarkAsync(telemetryContext, cancellationToken).
```

Comment:
Just checking, is this idempotent?  because you are not using state to keep track of whether the benchmark is downloaded. This is fine, just something to think about

Reply:
right now, I'm relying on blender benchmark itself to detect whether the blender engine and the scenes are installed already. Will open another PR if there are better ways to do this.